modules/compatibility/src/test/java/org/apache/ignite/compatibility/persistence/SnapshotCompatibilityTest.java

```diff
@@ -0,0 +1,459 @@
+
+        checkCaches(curIgn, cacheGrpInfo, BASE_CACHE_SIZE);
+    }
+
+    /** */
+    private void checkIncrementalSnapshot(IgniteEx curIgn) {
+        curIgn.snapshot().restoreSnapshot(SNAPSHOT_NAME, Collections.singleton(cacheGrpInfo.getName()), 1).get();
+
+        checkCaches(curIgn, cacheGrpInfo, BASE_CACHE_SIZE + ENTRIES_CNT_FOR_INCREMENT);
+    }
+
+    /** */
+    private void checkCacheDump(IgniteEx curIgn) throws IgniteCheckedException {
+        Map<String, Integer> foundCacheSizes = new ConcurrentHashMap<>();
+
+        Set<String> foundCacheNames = ConcurrentHashMap.newKeySet();
+
+        DumpConsumer consumer = new DumpConsumer() {
+            @Override public void start() {
+                // No-op.
+            }
+
+            @Override public void onMappings(Iterator<TypeMapping> mappings) {
+                // No-op.
+            }
+
+            @Override public void onTypes(Iterator<BinaryType> types) {
+                // No-op.
+            }
+
+            @Override public void onCacheConfigs(Iterator<StoredCacheData> caches) {
+                assertNotNull(cacheGrpInfo);
+
+                caches.forEachRemaining(cache -> {
+                    CacheConfiguration<?, ?> ccfg = cache.config();
+
+                    assertNotNull(ccfg);
+
+                    assertEquals(cacheGrpInfo.getName(), ccfg.getGroupName());
+
+                    foundCacheNames.add(ccfg.getName());
+                });
+            }
+
+            @Override public void onPartition(int grp, int part, Iterator<DumpEntry> data) {
+                assertNotNull(cacheGrpInfo);
+
+                data.forEachRemaining(de -> {
+                    assertNotNull(de);
+
+                    Integer key = (Integer)de.key();
+                    String val = (String)de.value();
+
+                    for (String cacheName : cacheGrpInfo.getCacheNamesList()) {
+                        if (val.startsWith(cacheName)) {
+                            assertEquals(calcValue(cacheName, key), val);
+
+                            foundCacheSizes.put(cacheName, foundCacheSizes.getOrDefault(cacheName, 0) + 1);
+
+                            break;
+                        }
+                    }
+                });
+            }
+
+            @Override public void stop() {
+                // No-op.
+            }
+        };
+
+        new DumpReader(new DumpReaderConfiguration(
+            CACHE_DUMP_NAME,
+            customSnpPath ? getCustomSnapshotPath(CUSTOM_SNP_RELATIVE_PATH, false) : null,
+            curIgn.configuration(),
+            consumer
+        ), log).run();
+
+        cacheGrpInfo.getCacheNamesList().forEach(
+            cacheName -> assertEquals(BASE_CACHE_SIZE, (int)foundCacheSizes.get(cacheName))
+        );
+
+        assertEquals(cacheGrpInfo.getCacheNamesSet(), foundCacheNames);
+    }
+
+    /** */
+    private @NotNull IgniteConfiguration getCurrentIgniteConfiguration(
```

Comment:
getCurrentIgniteConfiguration -> currentIgniteConfig

`get` must be omitted for internal code.